### PR TITLE
[feat] fetch interpretation listings from the cases table

### PR DESCRIPTION
**Pull request by ccatherinetan**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### :palm_tree: What's new in this PR
[//]: # "REQUIRED - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
* pulls case interpretation data from cases table (see `queries/languageSupport`) → merges case interpretation and non-case specific language support into a single array  
* displays the combined language support data on the `language-support` page (the case interpretations display at the bottom)
* exclusive case-interpretation fields: `upcoming_date, num_months, hours_per_month`
* exclusive non-case language-support fields: `deadline, num_pages` (i've included deadline for now, but we can always remove it later)
* all displayed fields are listed below. entries are "N/A" the field is not relevant to the case type 
id | title | summary | language | is_remote | listing_type | deadline | upcoming_date | num_months | hours_per_month | num_pages

Additionally:
* removes Immigration Law Enum from the app 

### :evergreen_tree: Screenshots
[//]: # "REQUIRED for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! If you are making changes here, please CC: @kyrenetam at the bottom."
Screen Shot 2024-03-07 at 9 37 56 PM
In this pic, the 2 bottom rows are case-interpretations. 

## :deciduous_tree: How to review
[//]: # 'REQUIRED - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
* check `app/language-support/page` for the page; `queries/languageSupport` for the case interpretation query
* nav to http://localhost:3000/language-support → compare against supabase entries 
* add a new entry into language-support table → check that it shows up on the page 
* add a new entry (with needs_interpreter = true) into the cases table + an entry into cases-languages, with the corresponding legal-server-id → check that the new case interpretation shows up on the page (should be at the bottom)

## :seedling: Next steps 
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
* style the language-support page 

### :information_source: Online sources
[//]: # 'Optional - copy links to any tutorials or documentation that was useful to you when working on this PR'
On using a union of prop interfaces: https://stackoverflow.com/questions/58630750/react-typescript-component-with-two-different-prop-interfaces 

### :potted_plant: Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # 'This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation.'
CC: @varortz

### CODE REVIEW

File: src/utils/helpers.ts [hunk 1]
```diff
@@ -68,21 +68,21 @@ export const isValidDate = (d: string) => {
...
-  if (experience === 'MEDIUM') {
-    return `${
-      experience.charAt(0).toUpperCase() + experience.toLowerCase().substring(1)
-    } (One prior case)`;
-  }
-  return `${
-    experience.charAt(0).toUpperCase() + experience.toLowerCase().substring(1)
-  } (Multiple prior cases)`;
-};
+// export const parseExperience = (experience: ImmigrationLawExperienceEnum) => {
```
Comment (varortz) [inline src/utils/helpers.ts]: since we don't need `immigration_law_experience` on the `Profile` schema, can we just delete these commented out lines of code (and anywhere else it was imported `CaseDetails` component i think) + remove the field from the schema

File: src/components/CaseDetails/index.tsx [hunk 2]
```diff
@@ -69,11 +69,11 @@ const caseFields = [
     label: 'Client Language',
     getValue: (data: CaseListing) => data.languages.join(', '),
   },
-  {
-    label: 'Attorney Experience Level',
-    getValue: (data: CaseListing) =>
-      parseExperience(data.experience_needed) || 'N/A',
-  },
+  // {
```
Comment (varortz) [inline src/components/CaseDetails/index.tsx]: let's remove this since we don't need it anymore

Comment (ccatherinetan) [reply src/components/CaseDetails/index.tsx]: sg!

File: src/utils/helpers.ts (hunk 1 above)
Comment (ccatherinetan) [reply src/utils/helpers.ts]: sg!

File: src/app/language-support/page.tsx [hunk 3]
```diff
@@ -29,6 +41,47 @@ export default function Page() {
     return 'null';
   };
 
+  const isCaseInterpretation = (obj: AllLanguageSupport) =>
```
Comment (varortz) [inline src/app/language-support/page.tsx]: Do you think we can update this so we check using the `listing_type` fields instead? I don't think the `CaseListing` schema currently has this field, so we'll need to add it! 

could we add a `listing_type` field to the schema? (this would be defaulted to "Case" in the `schema.ts` file). Once you pull the data for case-specific interpretations, we would want to change this field to "Case Interpretation".

Comment (varortz) [reply src/app/language-support/page.tsx]: we would need to add the `listing_type` field when pulling the data using with queries. (i.e. add the value "Case" for the `listing_type` field in the `getAllCases` query, and add the value "Case Interpretation" for the `listing_type` field in the `/queries/LanguageSupport.ts` query pulling case-specific interpretation)

Comment (ccatherinetan) [reply src/app/language-support/page.tsx]: done!

File: src/app/language-support/page.tsx [hunk 4]
```diff
@@ -37,25 +93,17 @@ export default function Page() {
...
-              <td>{l.title || 'null'}</td>
-              <td>{l.summary || 'null'}</td>
-              <td>{getIsRemoteValue(l.is_remote)}</td>
-              <td>{l.deadline}</td>
-              <td>{l.language}</td>
-              <td>{l.num_pages || 'null'}</td>
-            </tr>
-          ))}
-        </tbody>
+        <tbody>{allLanguageSupport.map(l => allLangSupportRow(l))}</tbody>
```
Comment (varortz) [inline src/app/language-support/page.tsx:106]: Looks like we're getting a console error from not having a key for all the mapped items, could we add the id as a key?)
image

**Review by varortz — APPROVED** [review]
LOOKS GOOD!! THANK U CATHERINE :))

File: src/utils/helpers.ts (hunk 1 above)
Comment (ccatherinetan) [reply src/utils/helpers.ts]: lemme check if i commented it out for the profile schema

File: src/app/language-support/page.tsx (hunk 3 above)
Comment (varortz) [reply src/app/language-support/page.tsx]: Actually I take this back sorry, after further discussion with @jinkang-0 (huge ++), this implementation would error because the type would be mismatched if we change to "Case Interpretation" since we have `listing_type: 'Case';` for `CaseListing`. Moving forward this is how we'll distinguish the types instead:
image

File: src/api/supabase/queries/languageSupport.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+ * @returns a Promise of all case interpretation objects
+ */
+export async function getCaseInterpretations(): Promise<CaseListing[]> {
+  const { data, error } = await supabase.rpc('getCasesData');
+  if (error) throw new Error(`Error fetching all cases: ${error.message}`);
+  const modifiedData = data
+    .filter((c: CaseListing) => c.needs_interpreter === true)
+    .map((caseInterpretation: CaseListing) => ({
+      ...caseInterpretation,
+      listing_type: 'Case Interpretation',
```
Comment (varortz) [inline src/api/supabase/queries/languageSupport.ts]: let's set `listing_type` to 'Case' so that it remains consistent with the `CaseListing` schema. See [this comment](https://github.com/calblueprint/immigration-justice-project/pull/66#discussion_r1520976926) for more info

File: src/api/supabase/queries/cases.ts [hunk 6]
```diff
@@ -5,7 +5,11 @@ import supabase from '../createClient';
 export async function getAllCases(): Promise<CaseListing[]> {
   const { data, error } = await supabase.rpc('getCasesData');
   if (error) throw new Error(`Error fetching all cases: ${error.message}`);
-  return data;
+  const modifiedData = data.map((item: CaseListing) => ({
+    ...item,
+    listing_type: 'Case',
```
Comment (ccatherinetan) [inline src/api/supabase/queries/cases.ts]: change to all caps "CASE"

File: src/app/language-support/page.tsx [hunk 7]
```diff
@@ -2,24 +2,43 @@
...
+import {
+  DocumentTranslation,
+  Interpretation,
+  CaseListing,
+} from '@/types/schema';
 
 export default function Page() {
-  const [docListings, setDocListings] = useState<DocumentTranslation[]>([]);
-  const [intListings, setIntListings] = useState<Interpretation[]>([]);
+  type AllLanguageSupport = DocumentTranslation | Interpretation | CaseListing;
```
Comment (jinkang-0) [inline src/app/language-support/page.tsx]: I think this would help for other parts of the app as well, can we put this in the schema?

File: src/app/language-support/page.tsx [hunk 8]
```diff
@@ -2,24 +2,43 @@
...
+            getAllInterpretation(),
+            getAllCases(),
+          ]);
+        setAllLanguageSupport([
+          ...docListings,
+          ...intListings,
+          ...casesInterpretationListings.filter(
+            caseInterpretation => caseInterpretation.needs_interpreter === true,
+          ),
+        ]);
```
Comment (jinkang-0) [inline src/app/language-support/page.tsx]: after adding the `upload_date` field, we can sort them here (and have case interpretations be at the top always)

Comment (ccatherinetan) [reply src/app/language-support/page.tsx]: done!

**Review by jinkang-0 — APPROVED** [review]
lgtm!
